lib/ansible/modules/cloud/amazon/ec2_instance_attribute_facts.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright, (c) 2018, Ansible Project
+# Copyright, (c) 2018, Moshe Immerman
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: ec2_instance_attribute_facts
+short_description: Get EC2 instance attribute facts
+description:
+    -  Get EC2 instance attribute facts
+version_added: 2.8
+author:
+    - Moshe Immerman <moshe.immerman@gmail.com>
+requirements: [ "boto3", "botocore" ]
+options:
+   instance_id:
+        description:
+            - EC2 Instance Id.
+        required: True
+   region:
+         description:
+            - AWS Region of the instance.
+         required: True
+   attribute:
+        description:
+            - The name of the instance attribute. See https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceAttribute.html
+        required: True
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = """
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+- ec2_instance_attribute_facts:
+    instance_id: "{{ec2_instance.id}}"
+    region: "{{region}}"
+    attribute: "userData"
+  register: userDataFact
+- debug: msg="{{userDataFact.attributes.UserData.Value}}"
+"""
+
+RETURN = """
+attributes:
+    description: A dictionary of instance attribute values.
+    returned: always
+    type: dict
+"""
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import ec2_argument_spec
+
+try:
+    import botocore.exceptions
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+
+def get_instance_attribute(connection, module):
+    try:
+        instance_id = module.params.get("instance_id")
+        attribute = module.params.get('attribute')
+        return connection.describe_instance_attribute(Attribute=attribute, InstanceId=instance_id)
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(e, msg="Couldn't get instance attributes")
+
+
+def main():
+    argument_spec = ec2_argument_spec()
```

Comment:
This is unnecessary, AnsibleAWSModule automatically merges the argument_spec that you pass with ec2_argument_spec().